codemod/src/plugin-deprecate/plugin-prop-rename.ts

```diff
@@ -0,0 +1,173 @@
+import type { PluginObj, PluginPass, Visitor } from '@babel/core';
+import { types as t } from '@babel/core';
+import type { NodePath } from '@babel/traverse';
+import { renderRecursiveDepthWarning } from '../warning-renderer/warning';
+import { deArray } from './helpers';
+
+interface Context extends PluginPass {
+  importNames: Map<string, string>;
+  namespace: string | null;
+  renames: Record<string, Record<string, string>>;
+}
+
+interface SubVisitorContext extends Context {
+  componentName: string;
+  propName?: string;
+  propLocation?: t.SourceLocation | null;
+  recurses: number;
+}
+const subVisitor: Visitor<SubVisitorContext> = {
+  JSXIdentifier(p) {
+    const newValue = this.renames[this.componentName]?.[p.node.name];
```

Comment:
Would this also rename a component called `bleedY` to `bleed`?  Probably not worth guarding against that case since it's quite unlikely.

Reply:
No i don't believe so. Its guarding against the prop look up on possible `undefined` (if there is no rename specified for the discovered component). We then check if there is a rename available just below it as well.

Happy to talk it through and add additional tests if you have an example in mind.

Reply:
Oh right, the prop name is scoped to a component. So you'd have to import `Button as bleedY` for it to do that.